src/app/api.ts

```diff
@@ -1082,6 +1108,96 @@ export function newAPI(): API {
     return secrets;
   }
 
+  async function createProjectSecret(
```

Comment:
looks like you aren't watching for prop changes. See other componentes like (projectsettings.vue watch method).

Should also add a test to check that this works.

Reply:
I tried to implement a watch function for the props changes and test it. Hope it's right.

Thank you.